server/src/main/java/org/apache/iotdb/db/engine/migration/MigrationTaskWriter.java

```diff
@@ -0,0 +1,173 @@
+      logger.error("unable to write to migrate log");
+    }
+  }
+
+  public void setMigration(MigrationTask migrationTask) throws IOException {
+    MigrationLog log =
+        new MigrationLog(
+            MigrationLog.LogType.SET,
+            migrationTask.getTaskId(),
+            migrationTask.getStorageGroup(),
+            migrationTask.getTargetDir().getPath(),
+            migrationTask.getStartTime(),
+            migrationTask.getTTL());
+    putLog(log);
+  }
+
+  public void startMigration(MigrationTask migrationTask) throws IOException {
+    MigrationLog log = new MigrationLog(MigrationLog.LogType.START, migrationTask.getTaskId());
+    putLog(log);
+  }
+
+  public void finishMigration(MigrationTask migrationTask) throws IOException {
+    MigrationLog log = new MigrationLog(MigrationLog.LogType.FINISHED, migrationTask.getTaskId());
+    putLog(log);
+  }
+
+  public void unsetMigration(MigrationTask migrationTask) throws IOException {
+    MigrationLog log = new MigrationLog(MigrationLog.LogType.CANCEL, migrationTask.getTaskId());
+    putLog(log);
+  }
+
+  public void pauseMigration(MigrationTask migrationTask) throws IOException {
+    MigrationLog log = new MigrationLog(MigrationLog.LogType.PAUSE, migrationTask.getTaskId());
+    putLog(log);
+  }
+
+  public void unpauseMigration(MigrationTask migrationTask) throws IOException {
+    MigrationLog log = new MigrationLog(MigrationLog.LogType.RESUME, migrationTask.getTaskId());
+    putLog(log);
+  }
+
+  public void error(MigrationTask migrationTask) throws IOException {
+    MigrationLog log = new MigrationLog(MigrationLog.LogType.ERROR, migrationTask.getTaskId());
+    putLog(log);
+  }
+
+  @Override
+  public void close() throws Exception {
+    logFileOutStream.close();
+  }
+
+  public static class MigrationLog {
+    public LogType type;
+    public long taskId;
+    public PartialPath storageGroup;
+    public String targetDirPath;
+    public long startTime;
+    public long ttl;
+
+    public MigrationLog() {}
+
+    public MigrationLog(LogType type, long taskId) {
+      this.type = type;
+      this.taskId = taskId;
+    }
+
+    public MigrationLog(
+        LogType type,
+        long taskId,
+        PartialPath storageGroup,
+        String targetDirPath,
+        long startTime,
+        long ttl) {
+      this.type = type;
+      this.taskId = taskId;
+      this.storageGroup = storageGroup;
+      this.targetDirPath = targetDirPath;
+      this.startTime = startTime;
+      this.ttl = ttl;
+    }
+
+    public MigrationLog(LogType type, MigrationTask task) {
+      this.type = type;
+      this.taskId = task.getTaskId();
+      this.storageGroup = task.getStorageGroup();
+      this.targetDirPath = task.getTargetDir().getPath();
+      this.startTime = task.getStartTime();
+      this.ttl = task.getTTL();
+    }
+
+    public enum LogType {
+      SET,
+      CANCEL,
+      START,
+      PAUSE,
+      RESUME,
+      FINISHED,
+      ERROR
+    }
```

Comment:
Maybe you can rename this enum to MigrationOperationType and make it an individual enum. So, you can use this enum to distinguish different operations.

Reply:
Changed.